apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -550,21 +536,21 @@ export const getResourceFullPermalink = async (
 
 interface PublishPageResourceArgs {
   logger: Logger<string>
-  user: User
+  userId: string
   siteId: number
   resourceId: string
+  sitePublishOptions:
+    | { enable: true; isScheduled: boolean; addCodebuildJobRow: boolean }
+    | { enable: false }
```

Comment:
question: why not make `sitePublishOptions` an optional prop then we don't need `enable`? we can just assume if prop is passed in, then it is enabled